classes/02class/exercises/c02-network12-challenge/nimmi89/README.md

```diff
@@ -0,0 +1,104 @@
+# c02-network12
+
+## Commands Execution Output
+
+For the exercises below, you should use all the custom created network resources. NOT THE DEFAULT ONES.
+
+- Provide commands used to go through all the below deliverables: (everything should be done using the AWS cli)
+
+  - explain how was the process to create a VPC peering
+  ```
+  The process of VPC peering involves the following steps:
+  >> Create a VPC Peering Connection using the requester VPC ID and the accepter Account ID / VPC ID
+  aws ec2 create-vpc-peering-connection --vpc-id vpc-081afcee350babdf5 --peer-vpc-id vpc-08fb60055e82efe81 --peer-owner-id 097922957316
+  >The command returns:
+  {
+  "VpcPeeringConnection": {
+      "AccepterVpcInfo": {
+          "OwnerId": "097922957316",
+          "VpcId": "vpc-08fb60055e82efe81",
+          "Region": "ap-southeast-2"
+      },
+      "ExpirationTime": "2020-07-13T09:18:50+00:00",
+      "RequesterVpcInfo": {
+          "CidrBlock": "172.16.0.0/16",
+          "CidrBlockSet": [
+              {
+                  "CidrBlock": "172.16.0.0/16"
+              }
+          ],
+          "OwnerId": "438549961569",
+          "PeeringOptions": {
+              "AllowDnsResolutionFromRemoteVpc": false,
+              "AllowEgressFromLocalClassicLinkToRemoteVpc": false,
+              "AllowEgressFromLocalVpcToRemoteClassicLink": false
+          },
+          "VpcId": "vpc-081afcee350babdf5",
+          "Region": "ap-southeast-2"
+      },
+      "Status": {
+          "Code": "initiating-request",
+          "Message": "Initiating Request to 097922957316"
+      },
+      "Tags": [],
+      "VpcPeeringConnectionId": "pcx-01cc84cf6f2956ae5"
+  }
+  }
+  >> Peer accepts the peering request
+  aws ec2 accept-vpc-peering-connection --vpc-peering-connection-id pcx-01cc84cf6f2956ae5
+
+  >>The SSH public key is shared to be used on the requester instance to connect via SSH
+  >>Both the acceptor and requester route tables are modified to allow target as peer VPC.
+  >>The NACL rule was allowed for the particular instance IP
+  >>The security group of the EC2 instance is updated to accept connection from specific IP.
+
+  ```
+
+  - add routes from origin VPC to target VPC - only for private subnets
+  ```
+  aws ec2 create-route /
+  --route-table-id rtb-068b99ba337982448 /
+  --destination-cidr-block 172.32.0.0/16 /
+  --vpc-peering-connection-id pcx-01cc84cf6f2956ae5
+
+  Similarly, for acceptor put your CIDR as destination.
+
+  ```
+  - open any network ACL to allow the connection within the subnets -- try to   only open the mininum required
+
+  ```
+  aws ec2 create-network-acl-entry \
```

Comment:
are you sure allowing ingress on port 22 is the only nacl required to get It working across both vpcs?

Reply:
The ephemeral port range would also be required for SSH return traffic. Have made the changes.